### DISCUSSION TITLE
Nav Bar / Menu items

### DISCUSSION THREAD

**Discussion by G-Ambatte** [naturalcrit/homebrewery#1210] [Ideas]
Noting down some of the ideas mentioned in Gitter chat today (25/01/21):

- addition of "New" item to navbar (see PR #1208)
- limited space on navbar means careful consideration is required for new navbar items, such as "Download source" (see PR #1198)

The discussion centres around the redesign of the nav bar system, including the potential creation of a new dropdown menu for the Edit page which contains the normal file operations - "new" "open" "print" "save" - although in Homebrewery's case, it may be more appropriate to shift the following existing navbar items:

- Report Issue
- Get PDF
- Share
- Save / Save to Google

**Comment by G-Ambatte** [comment]
From Gazook89:

> Snippet bar
Only for items that add code to the brew

> Top toolbar
For navigating the site....and maybe this includes the file menu with "new, save, create pdf, report issue, account" and even the metadata of a brew ("published, system, description, title")

**Reply by Gazook89** [reply]
We should consider and look into a zoom function for the preview window, adding that functionality to the UI.  I've done some work around that in a very wonky CSS way (resizing `.brewRenderer` at the same time as `.pages` to shrink one and grow the other simultaneously), but there has got to be better way.

The utility, for me at least, is being able to zoom out just enough to screenshot an entire page.  Or, to view a couple pages at a time (for which flexbox pages are handy, but that's a different thing).   Perhaps there simply needs to be a "fit page to preview window" option.

**Comment by Gazook89** [comment]
## Mock-up 1: Basic Menu

<details>
<summary>Click to show images</summary>

<img width="225" alt="image" src="https://user-images.githubusercontent.com/58999374/105659194-d0b73780-5e8d-11eb-81a8-5affb20ed29f.png">

<img width="225" alt="image" src="https://user-images.githubusercontent.com/58999374/105660815-7f10ac00-5e91-11eb-9f16-0390692a12c5.png">
</details>

Just a quick option for a drop down menu in the Editing page, and an option for the Share page  (Share menu was added later with some extra flourishes).  Matches the styling of the 'recent brews' menu.  I think it should be a "on click" action, rather than hover...and the recent brews should use onclick as well.   Placeholder icons used...could go without them as well.
### Pro's:
- simple to understand, simple to implement
- no need to dramatically change other styling
- easily expanded if more menu options needed.
### Cons:
- I personally don't like right-justified menus
- Brew Info option would likely require a pop up dialog box (that contains fields for Title, Description, etc).
- Very long account names might need to be cut off with ellipses `...`

**Reply by ericscheid** [reply]
In the application-space, the current left-most items should get priority (Naturalcrit, Homebrewery, version, help out) .. but in editing-space the editing functions should likely get priority .. and those app-items be relegated to a drop down menu, freeing up space.

Not sure what this would look like for the `/share/` page though. Still need a bunch of brew specific buttons, but likely also want to surface the app-specific items too (i.e. to recipients of the shared link).

**Reply by Gazook89** [reply]
I just added a "share page" menu as an edit to my post, same style with a couple of flourishes that better match the existing *recent brews* menu.  I also changed the original "Gazook89" at the bottom of the menu to "account" which I think is more descriptive.   This allows for the Share page menu to show a "Gazook89's Profile" menu option which would take a user to the brew author's page where they could view published brews.

One more 'con' is that the "Brew Info" option would likely need to open a dialog box...in my mind, it is where you would enter Title, choose tags, set published/private, etc.  From the share screen you would be able to read the description here.    To go further, I think the Title and Description could pop up on the Share screen as a dismissable 'notice'.

**Reply by ericscheid** [reply]
If we can pull together a comprehensive list of all the options we can do some card sort exercises.

**Reply by Gazook89** [reply]
You guys likely already have ideas on how to do a card sort, but some quick googling for myself turned up https://kardsort.com/ as a free and simple option.  I just ran their demo and seems simple enough.  It could be posted in the subreddit and maybe r/web_design or similar for more varied input.

**Comment by Gazook89** [comment]
## Mock-up 2: Floating Buttons

<details>
<summary> Click to show image </summary>
<img width="270" alt="image" src="https://user-images.githubusercontent.com/58999374/105671846-fbfb5000-5ea8-11eb-87cd-8d2c7b98c42b.png">  <img width="270" alt="image" src="https://user-images.githubusercontent.com/58999374/105672407-f8b49400-5ea9-11eb-844f-848b74e71a73.png">  
<img width="270" alt="image" src="https://user-images.githubusercontent.com/58999374/105672478-197ce980-5eaa-11eb-99ce-f5809ad0da99.png">
</details>

Buttons along the side.  Either in a strictly vertical format, or hidden groupings that expand out when their grouping button is clicked (grouping buttons not picture here).  One set has some colors just to see what it looks like...could be used as 'hover' or 'active' colors for the buttons.  The text labels were created just to provide context in this discussion, but on second look maybe they could be useful in production in the vertical format....slide the text out from behind the buttons as you hover?

Note, these are just mockups, not coded at all.

EDIT:
[Here is a codepen with some css as a base. ](https://codepen.io/Gazook89/pen/YzGoJvZ) I imagine that the more complex buttons would need to have `<img />` inside them.  I don't know if it makes any difference, but I created all of the icons and used SVGs to do so.

**Comment by G-Ambatte** [comment]
For the record, the current state of the nav bar is as below:

### Home:
- Report Issue
- Changelog
- Recent Brews
- Account

### New:
- Save
- Get PDF
- Report Issue
- Recent Brews
- Account

### Edit:
- Google Drive
- Save
- Report Issue
- Share
- Get PDF
- Recent Brews
- Account

### Share:
- Get PDF
- Source
- Recent Brews
- Account

### User:
- Recent Brews
- Account

### Print:
- no navbar

### Source:
- no navbar

**Reply by Gazook89** [reply]
So another way of looking at is as a table...maybe this is overkill, but helps me.  I listed the pages of the site, and the links needed, and then x'd for each link need for each page.  [I did this in a Google Sheet, and added a couple of comments on cells](https://docs.google.com/spreadsheets/d/1TV_F3THQnDF__iY98EieaojMJHU5BCQzv3OV_bhe3GE/edit?usp=sharing)....I hate to dilute conversation into yet another platform, but feel free to comment on it there or just here.

<details>
<summary>click to show table</summary>

  | NatCrit Landing Page | Homepage / root | New | Edit | Share | Source
-- | -- | -- | -- | -- | -- | --
Go to Homepage | x | x | x | x | x |  
Donate to NaturalCrit | x | x | x | x | x |  
Log In | x | x | x | x | x |  
Log Out | x | x | x | x | x |  
Account Settings | x | x | x | x | x |  
Change Log |   | x | x | x |   |  
Create New |   | x | x | x | x |  
View Owned Brews |   | x | x | x | x |  
View Recent Brews (owned & others) |   | x | x | x | x |  
Preview Pane View options |   | x | x | x | x |  
Add Snippets |   | x | x | x |   |  
Report Issue |   | x | x | x | x |  
View Share Link |   |   | x | x |   |  
Print/Create PDF |   |   | x | x | x |  
Download Owned Source |   |   | x | x |   |  
Edit Metadata |   |   | x | x |   |  
Toggle G Drive |   |   | x | x |   |  
Display Title (and author?) |   |   | x | x | x |  
View Source |   |   |   |   | x |  
View Edit Link |   |   |   |   | x |  
View Others Brews |   |   |   |   | x |  
View Metadata |   |   |   |   | x |  
</details>

**Comment by Gazook89** [comment]
## Mock-up 3: Two Full Width NavBars

<details>
<summary>Click to show image</summary>

<img width="930" alt="image" src="https://user-images.githubusercontent.com/58999374/105735649-8f5c7180-5ef9-11eb-904e-9c6e328f92e2.png">
</details>

Another option, expanding the Snippet bar to span the width of the entire window, rather than just the editor pane.  Means that it would need to adjust depending on the page (edit, share, etc).   

The save status of the brew is indicated next to the title, using `[M]` to show it is modified, not saved.  Similar to many programs using an asterisk.  I prefer the [m] as used in Affinity products as I think it is more visible.

This also shows buttons for zoom in, zoom out, and a menu for making some similar changes.  The 'card mode' and 'book spread' would need to be figured out so that their effects could be implemented on the fly, rather than making changes within the brew.   Card mode would use Flexbox to as many pages into a row as possible, and book spread would utilize CSS Grid and aim to fit two pages side by side (if iframe width allows it) and also start with the first page on the right side.

Finally, I changed the "Editor" snippet menu to "Layout"...I feel like Editor isn't very descriptive.  I think some of those snippets could be reorganized as well, but this and that discussion can be done elsewhere.

### Pros
- not very intrusive
- similar to other programs and their layered toolbars

### Cons
- moves some links/actions into drop down menus, adding another click to some actions
- as-is, might not as accessible as desired.  Low contrast between the light gray menu background and the editor pane./

**Comment by AlexeySachkov** [comment]
I think that we should start from the use-cases of each menu item in order to understand where to place it. And probably there should be more than just one menu.

A bit of more detailed explanation:

There should probably be some generic navbar, which belongs to the site itself and provides high-level navigation, like:
- project logo (which allows to go to a main page),
- current version, which points to change-log (just to look cool, I don't know; or that users could read about new features)
- account menu, which allows to open a user page
  - this could be a drop-down containing other account-related things like recent brews
- also I believe that a bug report button should be in that main menu as well and visible on all pages (as I stated here: https://github.com/naturalcrit/homebrewery/issues/1166#issuecomment-759399449).

This is probably all that I imagine at the moment for a site-wide menu, which is page-agnostic

Then there should be some second-level page-specific navbars, which are unique to each page to give quick access to the most needed feature. Basically, we only have three (but in fact four) different pages: home, edit, share and new (do we really need to distinguish it from the edit page?).

So, for edit page:
- Editor navbar with snippets, editor settings, etc.
- Brew navbar: a button to edit document metadata, save indicator
- View navbar: zoom, fit to screen, switch to a share mode, switch to print mode, switch to raw source mode

Home page:
The same as edit page, but no way to edit metadata, switch to a share or print mode

Share page:
- switch back to edit mode if you are an author
- switch to print mode
- switch to raw source mode

New page should probably be removed in favor of an edit page, but if not removed probably it should be the same as home page, except you can edit metadata

What do you think about that approach?

**Reply by Gazook89** [reply]
> Basically, we only have three (but in fact four) different pages: home, edit, share and new (do we really need to distinguish it from the edit page?).

I think we need a "new" page for unsaved brews...but if you are just wondering if there needs to be any functional/visual difference, then I agree we do not.

**Reply by G-Ambatte** [reply]
> new (do we really need to distinguish it from the edit page?)

Just a thought: what if rather than `/new/`, it was instead `/edit/` with no ID field, or an pre-defined pseudo-ID like `/edit/new`? On saving for the first time, the user is redirected to `/edit/{editID}`.
Then there really would be no difference between the edit and new pages.

**Reply by calculuschild** [reply]
There are a couple of fiddly things that have led to the separated pages. While they are 99% the same (and we can probably make up the differences just with an `if(editId)` to reduce redundancy)

### New Page:
1. Autosave is disabled. You have to manually click save at least once. This is partly to avoid lots of "empty" brews appearing in the database.
2. No Share link. Can't create a share link if the brew doesn't exist in the database yet.
3. No Google Drive toggle. Doesn't necessarily need to be missing, but that button does assume the brew already exists by the time you start clicking it. Logic would have to change.

**Reply by AlexeySachkov** [reply]
Thanks for info about `new` page, but I think whether to left it or not is a separate discussion.

My main point was to kind of think what do I want to do on particular page and based on that come up with particular nav items and their locations/grouping. For example I think that there should be no links to `user` page anywhere near dropdowns above the brew editor next to `snippets` or `table` nav items, like it is shown by @Gazook89 here: https://github.com/naturalcrit/homebrewery/discussions/1210#discussioncomment-309132. Just because user account doesn't below to brew editor.

Note: I specifically not talking about particular style and appearance, but only about location/grouping of those nav items by their purpose/area of effect.

**Comment by Gazook89** [comment]
Again, just another mockup (and I'm going to start labeling these to make it easier to discuss).
## Mock-up 4: Three Pane
<details>
<summary> Click to Show Image </summary>

<img width="850" alt="image" src="https://user-images.githubusercontent.com/58999374/106039277-92449700-609e-11eb-9603-7eaa383db09f.png">
</details>

I've sometimes wondered what a third pane would look like, such that css styling is done in one and the structure (md/html) in another.  One pain point when creating a whole new custom theme on the fly is knowing which styles already exist, and then adding code to override or remove those styles.  If the default stylesheet could be loaded into its own pane, and then modified and saved, that might be neat.  Saving the modified stylesheet might mean it is tied to particular brew, or saved as it's own thing but linked to the brew doc as the referenced stylesheet.  

***I'm going to revise this further, specifically the 'account' items, but posting this now because it's typed and I may have to jump ship abruptly before finishing revision***

**Comment by calculuschild** [comment]
FYI, @stolksdorf already laid the foundation in the V3  branch for a more sophisticated Editor panel. I'm pretty keen to follow his work that uses buttons to swap between three "tabs": 1) Markdown Editor, 2) CSS, 3) Metadata (which is currently the awkward separate popup window.  I am on board with adding additional buttons and rearranging the top nav bar/ editor panel, etc, but as for separate panel views I'm mostly planning to go with this approach (rather than a split view). Some screenshots taken from that version (remember that it is ~ 3years old and missing some of the current stuff):

Markdown panel. Note the three buttons on the right edge of the editor panel.
![image](https://user-images.githubusercontent.com/5878534/106041795-4566bd80-60aa-11eb-99fd-8a74886e7e4a.png)

CSS panel
![image](https://user-images.githubusercontent.com/5878534/106041886-5f080500-60aa-11eb-9cf1-21553ea279c6.png)

Metadata panel
![image](https://user-images.githubusercontent.com/5878534/106041949-6fb87b00-60aa-11eb-9c68-83b510a0eb29.png)

**Reply by Gazook89** [reply]
oh, sweet, and even has the separation of the "print" snippets into the CSS panel.  I didn't know about these plans already, and being able to switch full-height panels is much better than splitting limited real-estate.  

Edit: minor nitpick, but I think the hamburger icon for the metadata menu should be changed.  I don't know to what, but to me a hamburger icon is to indicate a dropdown menu of options, not really a dialog box for filling out info.

**Reply by calculuschild** [reply]
You can take a look at this Epic issue that will be the next major step after the v3 renderer drops. #1137 The idea is the contents of the CSS panel will be saved as a separate document, and the original Homebrew document will have a reference to it, so they load together. In this way, you can "share" your CSS around to other people without them having to copy and paste everything, or even just keep consistent custom styling between each of your own documents. (an update to font-size in my "Gazook89" style will update all of my documents that link to that. Hopefully, we could also have some "include" or "require" syntax in there to externally load or combine other additional CSS documents as well, including other custom "snippets" (Here's my custom "paper note" div. Import it into your project with "import(style/v1x_5xg)" or whatever.)

**Comment by 5e-Cleric** [comment]
Well, isn't this issue 3 years old now, youza! (as if i was here then, lol)

The current state of the navigation looks about the same:

<details>
<summary>Click to show image</summary>

![image](https://github.com/naturalcrit/homebrewery/assets/91662974/5aaa03fa-978c-40a8-be98-1d3f96c1f2b7)
</details>

With only 3 changes, New has a dropdown (From Blank / From File), The new page `Migrate` which helps users migrate brews from the legacy renderer into v3, and there is now an `account` page to show the details about the account.

[This option](https://github.com/naturalcrit/homebrewery/discussions/1210#discussioncomment-307734) by Gazook was great, not sure why we wouln't want to go that way:

<img width="170" alt="image" src="https://user-images.githubusercontent.com/58999374/105671846-fbfb5000-5ea8-11eb-87cd-8d2c7b98c42b.png">  <img width="170" alt="image" src="https://user-images.githubusercontent.com/58999374/105672407-f8b49400-5ea9-11eb-844f-848b74e71a73.png">  


Myself i like the idea, but we can't put that inside the brewRenderer component (see #3150).

**Reply by Gazook89** [reply]
I don’t think I envisioned it being IN the renderer. Just above

**Reply by Gazook89** [reply]
But also, I’m less fond of this idea now. It could swing back into my favor if I spent more time thinking about it, but I just haven’t done that in awhile.

**Comment by Gazook89** [comment]
Here is a comment on the /r/gmbinder subreddit, about why this particular user prefers GMB to Homebrewery.  I don't think it is necessarily due to any concerns about stability or features, but that aesthetics/UI are enough of a factor that they prefer GMB.  I largely agree with the points they raise.

-----

## From user
I prefer the more minimalist menu banner that GMBinder has:

-  it's got two less buttons than homebrewery, and could even lose another and still be as functional (the search and discover buttons are kind of redundant to each other).
-  I feel like the HB banner could be better condensed; the "NaturalCrit" link and "The Homebrewery" link are kind of redundant and the second could probably be removed without overly impacting the user experience.
-   I would swap the "Need Help?" button into a submenu under profile for a dedicated "Brews" button, "Recent Brews" isn't useful enough for me to justify a dedicated button in my experience
- I also don't feel like the changelog is deserving of a banner link, I'd bundle it under the profile button as well. Yeah it's not profile related, but I don't feel like it would stick out as overly out of place if it got thrown above the logout button
- The brew view is also unappealing to me. GMB has rows of documents with a document icon, title, modified date, published tag, and "menu" button. HB is organized in a tile format, which is great for tablet's but I don't foresee myself doing any work on a tablet, I'm actually not sure if that's an intended or developed feature at this time.
-  Additionally the physical separation of "published" and "unpublished" is unnecessary and clutters the page, there's already a small icon for HB-local-storage or cloud synced, another icon for published or unpublished would be more aesthetically pleasing
-  Having a link to the users profile on each brew, while looking at their brewery page, is redundant and unnecessary, especially since it links back to the page you're currently on. I imagine this is for more simplified organization on the back-end, as it looks like the vault is setup with the same organizational structure, which would be more useful since you're not looking at the same users content, but is unnecessary when looking at a users brewery
-  To open a specific brew I have to hover the tile and then click the edit button, it's more intuitive to me to click the title of the brew (or even the tile itself) to open the editor rather than having to click a specific button that pops up in a secondary menu

I was going to go into more about the editor, but it seems to have been updated a fair amount since I last used the site to any significant amount and seems a lot more fleshed out with utilities.

So really at this point I think it just comes down to aesthetics and efficiency of linkage/organization for me, which is of course up to the end-user and developers can't please everyone.

And for context; I'm not a power user, I've got maybe 15 to 20 documents I have in GMB, less than half of which are over 10 pages long, I'm not using it with any plans to publish anything, it's entirely just for my own organization. I'm also not really in a financial situation to support the project, nor have I taken part in any feedback on the site/forums, so on the sliding scale of priorities I feel like my opinion is (deservedly) negligible to the developers. 

-------

**Reply by Gazook89** [reply]
To this end, here is what I have in a local branch right now (`/edit/` page):

<img width="1540" alt="image" src="https://github.com/user-attachments/assets/dda0ef13-1b1f-49f4-af39-7aa5ea97eb1d" />

<img width="1540" alt="image" src="https://github.com/user-attachments/assets/866554fd-cfbe-4701-8d98-fd7ea354a6db" />

<img width="1540" alt="image" src="https://github.com/user-attachments/assets/002055ed-5868-4f44-9372-7d6e4a4051e6" />

Since I moved some dropdown menus into other dropdown menus ("new", and likely "recent brews"), I have to get that to work nicely still.  Though, mostly for Firefox's benefit, since with Chrome it'll be easy with the help of Anchor and Popup APIs.

And add little dividers within dropdown menus to create groups (likely just a slightly thicker line).

**Reply by Gazook89** [reply]
Oh and in the process I combined `shared/.../nav.jsx` with `client/.../navbar.jsx` because the separation has long bothered me, and made it all a functional component as well as each of the `.navitem.` files.

**Reply by 5e-Cleric** [reply]
I do agree with this point, but the merge may be too much, also, the homebrewery link i'd replace with a `home` icon, as per usual on sites, or a proper logo.

AS per the rest, i have a few points:

1- keep the account stuff at the top right, that bit is basic in 90% of websites.
2- Separate the brew dropdown into routes and brew stuff? like: share brew, new brew, import instead of new from file, and then the brew stuff with save, print, post to reddit
3- not sure about autosave there, but if it is, autosaved and the autosave should be the same element.

**Comment by Gazook89** [comment]
Just a place to drop a note and link....

We should allow users to remove the `trashed` property from Google Drive brews if they are found in the trash, rather than just telling them that it's in the trash.  Here is a google drive api doc for it: https://developers.google.com/workspace/drive/api/guides/delete   (tl;dr, it requires an `update` call with the `trashed` property set to false.

~~Also, currently, we only give users this notice if they go into a brew to edit it-- but it'd be valuable if they could see that in the brew list page even if they can't act on it there.~~

**Reply by calculuschild** [reply]
> it'd be valuable if they could see that in the brew list page even if they can't act on it there.

We explicitly hide files in the trash folder because otherwise deleted files would continue to be displayed in their User Brews page. I imagine people would be very confused if they delete a Google brew only to see it's still showing up in their list.

**Reply by Gazook89** [reply]
currently we have a "published" and "unpublished" distinction, I don't think it's much of a stretch to have a "trashed" section.   

But so right now, is the Trashed notification that we pop just for if a person follows an old link to their brew and it's trashed?

**Reply by calculuschild** [reply]
Right. If a user deletes their brew, we could either permanently delete, or just drop in the Google Drive trash. We opted to put in the trash so they have an opportunity to retrieve it if they change their mind. But for the purposes of The Homebrewery, we assume it was an intentional deletion and no longer attempt to search for it or display it.

If a user visits a trashed brew via an old link, we warn them so they aren't mistakenly putting work into a document that Google is about to clean out.

**Reply by calculuschild** [reply]
If we add a "trashed" section, I expect users will be further confused that only some brews land there (Google Brews) while others don't (HB Brews). And then of course the complaints "why do my deleted brews not actually delete but are just moved to a Trash folder? How do I set it to just fully delete?" Not saying we can't add it but I think it would need to be universal to all brews somehow.

Have you actually seen any requests from users wanting to keep their deleted documents visible on their user page?

**Reply by Gazook89** [reply]
fair enough.  strike that part about the user page

**Reply by 5e-Cleric** [reply]
But, isn't the case different if the user mistakenly deleted the brew from their drive directly rather than purposefully deleting them using our UI? Should our response be the same?

**Reply by calculuschild** [reply]
> if the user mistakenly deleted the brew from their drive directly

Hm... I'm not sure. I don't know how we would differentiate between files deleted via HB and deleted via Drive. Or accident vs intentional for that matter.

If we wanted to alert them somehow, would that be annoying every time you delete a file, HB sends a popup "we found brews in your trash!" I don't know.

Would users think to check the Trash folder on their own without an alert? Is the warning on the edit page enough? Is this an active concern from users?

**Reply by 5e-Cleric** [reply]
uhm, so just deleting the stub or not deleting the stub.

If the brew is deleted on site, we delete the stub, or we apply a property to the stub, if it is deleted elsewhere, the stub won't have that property, doesn't seem that difficult from here.

**Comment by Gazook89** [comment]
The 'Transfer to Google Drive/HB' button should open a dialog or modal, rather than just doing it, so that we can display more information about what the button is actually doing.  A dialog/modal would also give us space to show any resulting error (or success state) directly within the same space before closing the modal.  And, if we ever have more storage services, multiple services could be displayed rather than be a 'on/off' toggle for one service.

**Reply by calculuschild** [reply]
It does open up a descriptive popup that the user has to confirm before the transfer occurs.

![image](https://github.com/user-attachments/assets/bafb45bd-b049-44a8-b520-554943779c4a)

**Comment by Gazook89** [comment]
I took down my UI Overhaul PR #4122  because it was so stale, too large, and didn't garner much feedback.  But I still have it on my local.  Just grabbing some additional screenshots that didn't make it into the PR.

Navigation bar ideas:

<img width="436" height="287" alt="image" src="https://github.com/user-attachments/assets/8438409c-772f-4dbb-8972-7e59d7fb7b82" />

<img width="436" height="287" alt="image" src="https://github.com/user-attachments/assets/547a908b-fb41-4121-ad98-c59970610020" />

<img width="436" height="287" alt="image" src="https://github.com/user-attachments/assets/b054b944-92ea-4765-95c2-9837cf0cafec" />

<img width="436" height="287" alt="image" src="https://github.com/user-attachments/assets/20a3673f-5147-4d1d-a242-a0cab7b19c41" />

<img width="436" height="287" alt="image" src="https://github.com/user-attachments/assets/15f98ff6-e1ca-40d4-afa6-1ca99d06c6a8" />

**Reply by 5e-Cleric** [reply]
I like this, might not put the menus in the same order but its miles better than what we have.

**Reply by Gazook89** [reply]
> I like this, might not put the menus in the same order but its miles better than what we have.

I just followed most file menus:

- New
- save (not evident here)
- open
- export
- print

Off the top of my head I'm not sure I remember which is first between Save and Open but easy swap.

**Reply by 5e-Cleric** [reply]
I'd go with:

- moving patreon higher, and changelog lower
- Adding a few more icons, `?` to Faq, `fas_house_chimney` for home

and maybe adding padding, but that's me, i have big screens, so lots of space.
